### PR TITLE
re-indent the code and delete some comments

### PR DESCRIPTION
**Pull request by ruqiluo**

### CODE REVIEW

File: src/least_square.cc [hunk 1]
```diff
@@ -28,330 +25,335 @@
...
+                                     Vector&,
+                                     Parameters&);
 
 /**********************************************************************
  *
  * Calculates the objective value for l1 regularized least square.
  * Input:
- *     A:      the data matrix with size num_features x num_samples;
- *     (type T, it can be sparse matrix SpMat or Matrix)
+ *     A:      the data matrix with size num_features x num_samples; //======Ruqi: Does this 'x' mean '*'?======
```
Comment (ZhiminPeng) [inline src/least_square.cc]: yes, change "x" to "*", thanks.

File: src/least_square.cc [hunk 2]
```diff
@@ -28,330 +25,335 @@
...
-  sub(grad_loss, b);  // calculates the gradient of the loss function Atx-b
-  double nrm_grad_loss = norm(grad_loss, 2);
-  double nrm_x = norm(x, 1);
-  return lambda * nrm_x + 0.5 * nrm_grad_loss * nrm_grad_loss;
+double l1_objective(T&          A,
+                    Vector&     b,
+                    Vector&     x,
+                    Vector&     Atx,
+                    Parameters& para)
+{
```
Comment (ZhiminPeng) [inline src/least_square.cc]: { goes to line 88.

File: src/least_square.cc [hunk 3]
```diff
@@ -28,330 +25,335 @@
...
-      local_end = (block_id + 1) * block_size;
-      if(block_id == num_blocks-1) {
+template <typename T>
+void l2_ls(T&          A,
+           Vector&     b,
+           Vector&     x,
+           Vector&     Atx,
+           Vector&     Ab,
+           Parameters& para)
+{
```
Comment (ZhiminPeng) [inline src/least_square.cc]: { goes to line 142.

File: src/least_square.cc (hunk 2 above)
Comment (ruqiluo) [inline src/least_square.cc]: Yes! What would you like it to be?

File: src/least_square.cc [hunk 4]
```diff
@@ -28,330 +25,335 @@
...
+    int num_blocks = num_features / block_size;
+    int i;
+    int block = 0;
+    int local_num_blocks = num_blocks / num_threads;
+    int block_id;
+    Vector local_dx(num_features - (num_blocks - 1) * block_size, 0.);
+    // main loop; each iteration represents an epoch
+    for (int itr = 0; itr < MAX_ITER; itr++) {
+        for (block = 0; block < local_num_blocks; block++) {
+            // generate a random block id
```
Comment (ZhiminPeng) [inline src/least_square.cc]: add step 1 to here.

File: src/least_square.cc [hunk 5]
```diff
@@ -28,330 +25,335 @@
...
+            local_start = block_id * block_size;
+            local_end = (block_id + 1) * block_size;
+            if (block_id == num_blocks - 1) {
+                local_end = num_features;
+            }
+            // clean the data in local_delta_Atx
+            local_dAtx.setZero();
+            // local for loop for each thread
+            for (i = local_start; i < local_end; i++) {
+                // Step 1. generate a random index   //======Ruqi: Maybe step 1 is to assign index?======
```
Comment (ZhiminPeng) [inline src/least_square.cc]: remove this line

File: src/least_square.cc [hunk 6]
```diff
@@ -28,330 +25,335 @@
...
+            // output from thread 0.
+            cout << l1_objective(A, b, x, Atx, para) << endl;
+        }
     }
-  }
-
-  if(my_rank == 0 && flag) std:cout<<"];"<<endl;
-  return;
+    
+    if (my_rank == 0 && flag){
```
Comment (ZhiminPeng) [inline src/least_square.cc]: need a space between ) and {.

**Pull request merged**